src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -60,6 +63,17 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
         if ((child == null || child.asRustPath != null) && path.asRustPath == null) {
             holder.createErrorAnnotation(path, "Invalid path: self and super are allowed only at the beginning")
         }
+
+        path.self ?: return
+        val function = path.parentOfType<RsFunction>() ?: return
+        val rustPath = path.asRustPath
+        if (rustPath != null && rustPath !is RustPath.ModRelative && function.isAssocFn) {
+            val error = "The self keyword was used in a static method [E0424]"
+            val annotation = holder.createErrorAnnotation(path, error)
+            if (function.role != RsFunctionRole.FREE) {
```

Comment:
@matklad, it it intentional? I would expect `fn.isAssocFn` to return `false` if it's not an impl or a trait function.